### PR TITLE
Add flannel task

### PR DESCRIPTION
**Pull request by stgleb**

### CODE REVIEW

File: pkg/runner/ssh/runner.go [hunk 1]
```diff
@@ -74,7 +74,7 @@ func (r *Runner) Run(cmd *runner.Command) (err error) {
 	defer session.Close()
 
 	session.Stdout = cmd.Out
-	session.Stderr = cmd.Err
+	session.Stderr = cmd.Out
```
Comment (saheienko) [inline pkg/runner/ssh/runner.go]: Why did you change it?

File: scripts/flannel.sh.tpl [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+mv /usr/bin/flanneld-{{ .Arch }} /usr/bin/flanneld
+chmod 755 /usr/bin/flanneld
+sed -i 's/REPLACEME/'`ifconfig|grep "10\."|grep "inet "|cut -f10 -d" "`'/g' /etc/default/flanneld
+
+echo "[Unit]
+Description=Networking service
+Requires=etcd-member.service
+[Service]
+Environment=FLANNEL_IMAGE_TAG=v{{ .Version }}
+ExecStartPre=/usr/bin/etcdctl set /coreos.com/network/config '{"Network":"{{ .Network }}", "Backend": {"Type": "{{ .NetworkType }}"}}'" > \
```
Comment (saheienko) [inline scripts/flannel.sh.tpl]: Double quotes in the json config will be suppressed by the shell:
````
$ echo "[Unit]
Description=Networking service
Requires=etcd-member.service
[Service]
Environment=FLANNEL_IMAGE_TAG=v{{ .Version }}
ExecStartPre=/usr/bin/etcdctl set /coreos.com/network/config '{"Network":"{{ .Network }}", "Backend": {"Type": "{{ .NetworkType }}"}}'"

[Unit]
Description=Networking service
Requires=etcd-member.service
[Service]
Environment=FLANNEL_IMAGE_TAG=v{{ .Version }}
ExecStartPre=/usr/bin/etcdctl set /coreos.com/network/config '{Network:{{ .Network }}, Backend: {Type: {{ .NetworkType }}}}'
````

File: pkg/jobs/flannel.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	"io"
+	"text/template"
+
+	"github.com/pkg/errors"
+
+	"github.com/supergiant/supergiant/pkg/runner"
+	"github.com/supergiant/supergiant/pkg/runner/ssh"
+)
+
+type FlannelJobConfig struct {
```
Comment (saheienko) [inline pkg/jobs/flannel.go]: It would be better to move this package under pkg/jobs/flannel to name this struct as Config.

File: scripts/flannel.sh.tpl (hunk 2 above)
Comment (saheienko) [reply scripts/flannel.sh.tpl]: Better to use cat in this case:
```
cat << EOF   
heredoc> [Unit]
Description=Networking service
Requires=etcd-member.service
[Service]
Environment=FLANNEL_IMAGE_TAG=v{{ .Version }}
ExecStartPre=/usr/bin/etcdctl set /coreos.com/network/config '{"Network":"{{ .Network }}", "Backend": {"Type": "{{ .NetworkType }}"}}'
EOF
```

File: pkg/jobs/flannel.go (hunk 3 above)
Comment (stgleb) [reply pkg/jobs/flannel.go]: Ok

File: pkg/jobs/fake_runner.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+	"errors"
+	"io"
+	"strings"
+
+
+	"github.com/supergiant/supergiant/pkg/runner"
+)
+
+type FakeRunner struct {
+	ErrMsg string
```
Comment (saheienko) [inline pkg/jobs/fake_runner.go]: Why do you you an error message instead of an error?

File: pkg/jobs/fake_runner.go [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import (
+	"errors"
+	"io"
+	"strings"
+
+
+	"github.com/supergiant/supergiant/pkg/runner"
+)
+
+type FakeRunner struct {
```
Comment (saheienko) [inline pkg/jobs/fake_runner.go]: Should it be part of the pkg/jobs package?

File: pkg/jobs/flannel/flannel_test.go [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+/etc/systemd/system/flanneld.service
+systemctl enable flanneld.service
+systemctl restart flanneld.service
+`
+
+	tpl, err := template.New("").Parse(script)
+
+	if err != nil {
+		t.Errorf("error parsing flannel test template %s", err.Error())
+		return
```
Comment (saheienko) [inline pkg/jobs/flannel/flannel_test.go]: You can use [t.Fatalf()](https://golang.org/pkg/testing/#T.Fatalf) instead of t.Errorf() followed by return statement.

File: pkg/jobs/fake_runner.go (hunk 5 above)
Comment (egopher) [reply pkg/jobs/fake_runner.go]: +1 Should be part of testutils

File: pkg/jobs/fake_runner.go (hunk 4 above)
Comment (stgleb) [reply pkg/jobs/fake_runner.go]: I've found it easier to compare strings in tests instead of errors.

File: pkg/jobs/flannel/flannel_test.go (hunk 6 above)
Comment (stgleb) [reply pkg/jobs/flannel/flannel_test.go]: @saheienko ok

File: pkg/jobs/fake_runner.go (hunk 4 above)
Comment (saheienko) [reply pkg/jobs/fake_runner.go]: It is a bad practice, because different errors can have the same message.

Comment (stgleb) [reply pkg/jobs/fake_runner.go]: @saheienko Ok. I've changed that.

File: pkg/jobs/flannel/flannel.go [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+	}, nil
+}
+
+func (i *Job) InstallFlannel(config JobConfig) error {
+	err := jobs.RunTemplate(context.Background(), i.scriptTemplate, i.runner, i.output, config)
+	if err != nil {
+		return err
+	}
+
+	return nil
```
Comment (saheienko) [inline pkg/jobs/flannel/flannel.go]:
```
+	if err != nil {
+		return err
+	}
+
+	return nil
```
"gofmt -s" recognizes such blocks of code.
